frontend/types/campaignCharities.ts

```diff
@@ -14,8 +15,8 @@ export type CampaignCharityData = CampaignCharityBaseData & {
   charity: CharityData;
 };
 
-export type CampaignCharityDonationPublicData = CampaignCharityData & DonationBreakdownData;
+export type CampaignCharityDonationPublicData = CampaignCharityData & DonationBreakdownData & PrimaryDonorData;
 
-export type CampaignCharityDonationData = CampaignCharityData & DonationBreakdownData;
+export type CampaignCharityDonationData = CampaignCharityData & DonationBreakdownData & PrimaryDonorData;
```

Comment:
Doesn't some fields clash. ID in CampaignCharityData and PrimaryDonorData?

Reply:
Meant for PrimaryDonorData to be in a nested object `primaryDonor`, fixed